kedro/contrib/config/templated_config.py

```diff
@@ -0,0 +1,185 @@
+"""This module provides ``kedro.config`` with the functionality to load one
+or more configuration files from specified paths, and replace template strings with default values.
+"""
+import re
+from typing import Any, Dict, Iterable, Optional, Union
+
+import jmespath
+
+from kedro.config import ConfigLoader
+
+
+class TemplatedConfigLoader(ConfigLoader):
+    """
+    Extension of the ConfigLoader class that allows for template values, wrapped in brackets like:
+     ${..}, to be replaced by default values.
+
+    The easiest way to use this class is by incorporating it into the KedroContext. This can be done
+    by extending the KedroContext and overwriting the config_loader method, making it return an
+    TemplatedConfigLoader object instead of a ConfigLoader object.
+
+    For this method to work, the context_path variable in the .kedro.yml needs to be pointing at
+    this newly created class. The run.py script has an extension of the KedroContext by default,
+    called the ProjectContext.
+
+    Example:
+    ::
+        >>> from kedro.context import KedroContext, load_context
+        >>> from kedro.contrib.config import TemplatedConfigLoader
+        >>>
+        >>> class MyNewContext(KedroContext):
+        >>>
+        >>>    @property
+        >>>    def config_loader(self) -> TemplatedConfigLoader:
+        >>>        conf_paths = [
+        >>>            str(self.project_path / self.CONF_ROOT / "base"),
+        >>>            str(self.project_path / self.CONF_ROOT / self.env),
+        >>>        ]
+        >>>        return TemplatedConfigLoader(conf_paths,
+        >>>                                     globals_pattern="*globals.yml",
+        >>>                                     globals_dict={"param1": "CSVLocalDataSet"})
+
+        >>> my_context = load_context(Path.cwd(), env=env)
+        >>> my_context.run(tags, runner, node_names, from_nodes, to_nodes)
+    """
+
+    # pylint: disable=missing-type-doc
+    def __init__(
+        self,
+        conf_paths: Union[str, Iterable[str]],
+        *,
+        globals_pattern: Optional[str] = None,
+        globals_dict: Optional[Dict[str, Any]] = None
+    ):
+        """Instantiate a TemplatedConfigLoader.
+
+        Args:
+            conf_paths: Non-empty path or list of paths to configuration
+                directories.
+            globals_pattern: Optional keyword-only argument specifying a glob pattern.
+                Files that match the pattern will be loaded as a dictionary with default
+                values used for replacement.
+            globals_dict: Optional keyword-only specifying an additional dictionary with
+                default values used for replacement. This dictionary will get merged with
+                the globals dictionary obtained from the globals_pattern. In case of
+                duplicate keys, the arg_values keys take precedence.
+        """
+
+        super(TemplatedConfigLoader, self).__init__(conf_paths)
```

Comment:
```suggestion
        super().__init__(conf_paths)
```
For consistency with `get()` below :)